palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/response/AuditMessage.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2019 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.palisade.response;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.util.Assert;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.Generated;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+
+/**
+ * Audit information for a request provided by each service.
+ * Each individual service sends a record to the Audit Service for every request that it receives.
+ * The components of the message will differ depending on which service has sent the data and if the processing was
+ * successful or not.
+ *
+ */
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public class AuditMessage {
+
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public final String timeStamp; //when the service processed the request
+    public final String serverIp;   //server for the service that processed the request
+    public final String serverHostname;  //server host name for the service
+    public final Context context;  //Context of the client's request
+    public final String userId;    //User Id for the client.  Can be null if there is a User
+    public final String resourceId;  //Resource Id for the client.  Can be null if there is a Resource
+    public final String errorMessage;  //Error message that occurred during the processing of the request.  Will be null if there was no issue.
+
+    @SuppressWarnings("checkstyle:parameterNumber")
+    @JsonCreator
+    private AuditMessage(
+            final @JsonProperty("timeStamp") String timeStamp,
+            final @JsonProperty("serverIp") String serverIp,
+            final @JsonProperty("serverHostname") String serverHostname,
+            final @JsonProperty("context") Context context,
+            final @JsonProperty("userId") String userId,
+            final @JsonProperty("resourceId") String resourceId,
+            final @JsonProperty("errorMessage") String errorMessage) {
```

Comment:
All the same comments as the audit-service/AuditMessage
This should however only have the request was successful/request was error variants, it won't need the read* ones

Reply:
Complete overhaul of AuditMessage. Have another look.